src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -593,4 +598,26 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
                     .body(null);
         }
     }
+
+    /**
+     * GET /programming-exercises/:exerciseId/is-released-and-has-results : Check if the given exercise is released and has at least one student result.
+     *
+     * @param exerciseId the id of a ProgrammingExercise
+     * @return the ResponseEntity with status 200 (OK) and true if the exercise is released and there is at least one result, false if not. Returns 404 (notFound) if the exercise does not exist.
+     */
+    @GetMapping(value = "/programming-exercises/{exerciseId}/is-released-and-has-results")
```

Comment:
I am not sure if this is a good REST URL?!

Reply:
I'm not sure what the best convention is to follow here. I would argue that it we don't load the results from the db here and send them to the client but instead only perform an `exist` operation on the db that should be more efficient.